projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -977,15 +977,18 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
     }
 
     protected onStatusChanged() {
-        if ((this._ngControl.control.touched || this._ngControl.control.dirty) &&
-            (this.inputDirective && this._ngControl.control.validator || this._ngControl.control.asyncValidator)) {
-            if (this.inputGroup.isFocused) {
-                this.inputDirective.valid = this._ngControl.valid ? IgxInputState.VALID : IgxInputState.INVALID;
+        if (this.inputDirective) {
+            if ((this._ngControl.control.touched || this._ngControl.control.dirty) &&
+                (this.inputDirective && this._ngControl.control.validator || this._ngControl.control.asyncValidator)) {
```

Comment:
do not need to check for `inputDirective` again here - at this point it is truthy.